chainer/dataset/tabular/join.py

```diff
@@ -0,0 +1,54 @@
+import six
+
+from chainer.dataset import TabularDataset
+
+
+class Join(TabularDataset):
+
+    def __init__(self, *datasets):
+        keys = set(datasets[0].keys)
+        for dataset in datasets[1:]:
+            if not len(dataset) == len(datasets[0]):
+                raise ValueError('All datasets must have the same length')
+            if len(keys.intersection(dataset.keys)) > 0:
+                raise ValueError('All keys must be unique among all datasets')
+            keys = keys.union(dataset.keys)
+
+        self._datasets = datasets
+
+    def __len__(self):
+        return len(self._datasets[0])
+
+    @property
+    def keys(self):
+        return tuple(key for dataset in self._datasets for key in dataset.keys)
+
+    @property
+    def mode(self):
+        return self._datasets[0].mode
+
+    def get_examples(self, indices, key_indices):
+        if key_indices is None:
+            key_indices = six.moves.range(len(self.keys))
```

Comment:
I think we can directly return results of `get_examples(indices, None)` if `key_indices is None`.

```suggestion
            examples = []
            for d in self._datasets:
                examples.extend(d.get_examples(indices, None))
            return tuple(examples)
```

Reply:
I updated in #7116